src/ai/core/signal_processor.py

```
@@ -0,0 +1,172 @@
+    intent: float = 0.35
+    market: float = 0.25
+    regional: float = 0.20
+    trends: float = 0.20
+
+class SignalProcessor:
+    def __init__(self, weights: Optional[SignalWeights] = None):
+        self.weights = weights or SignalWeights()
+
+    def calculate_weighted_scores(
+        self,
+        property_data: Dict,
```

Ruff (0.8.2)

Unused method argument: `property_data`

(ARG002)